### ISSUE TITLE
Missing factor 0.5 in the VAE loss

### ISSUE THREAD

**Issue by msseibel** [ABotond/SD-LayerNet#2]
Hi Botond,

going through the code, I believe there is a minor mistake for the VAE loss in this line:
`kl_i = 1.0 + z_logvar - (z_mean ** 2) - torch.exp(z_logvar)`
https://github.com/ABotond/SD-LayerNet/blob/68dc3bd4073860a313687e73aad0ad68784091e0/model.py#L89C1-L89C68

To my understanding, the last term should be `torch.exp(0.5*z_logvar)`. In the ModalityEncoder itself, the factor is included.

Best,
Marc

**Comment by msseibel** [comment]
Never mind, I was wrong. The VAE loss uses the variance and not the standard deviation.
